src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,57 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidInputDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserNotNull(user);
+        checkUserNotRegisteredYet(user);
+        checkUserAge(user);
+        checkUserLoginLength(user);
+        checkUserPasswordLength(user);
```

Comment:
I'd move the already registered check to the end of the method, because it is an expensive operation and it is better to do it after all of the cheap operations are done